pkg/reconciler/route/resources/ingress_test.go

```diff
@@ -46,6 +47,51 @@ func getServiceVisibility() sets.String {
 	return sets.NewString()
 }
 
+func TestMakeIngress_CorrectMetadata(t *testing.T) {
+	targets := map[string]traffic.RevisionTargets{}
+	ingressClass := "foo-ingress"
+	r := &v1alpha1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-route",
+			Namespace: "test-ns",
+			Labels: map[string]string{
```

Comment:
Do we care to add annotations here as well, to test if they are propagated as well?

We could also enhance this test a bit to make sure that the predefined labels override anything that is provided via the route (for correctness). In the annotations case, we could actually check as well that a provided class annotation is **not** overridden but passed down correctly.

Reply:
Yes, sounds great. I added the enhancement as well as annotations to make sure.